### DISCUSSION TITLE
await set

### DISCUSSION THREAD

**Discussion by misaku** [pmndrs/zustand#804] [Q&A]
how can i await to chage value when ai use action to set value?

**Comment by dai-shi** [comment]
Can you provide a small code snippet?

**Comment by misaku** [comment]
yes i made one solution but i think it is not a goot idea.
```TS
const forceAwait = async (get: () => any, name: string, value: any) =>
  new Promise((resolve) => {
    let data: any;
    do {
      data = get()[name];
      if (data === value) resolve(true);
    } while (data !== value);
  });
  
  //IMPLEMENTATION
  
const useRodneyDataStore = create<RodneyBroadcastContextDataStore>(
    (set, get) => ({
      data: null,
      setData: async (value, callback) => {
        set({ data: value });
        await forceAwait(get, 'data', value);
        if (callback) await callback(get().data);
      },
    })
  );
  
  // now i can use this
  setData: useRodneyDataStore((state) => state.setData),
  await setData('xyz')
  // or
  setData('xyz', async()=>{
   // dothis
  })
```

**Reply by misaku** [reply]
i made it because some times i need to clear global state before change route, and i cant use useEffect to do this

**Reply by dai-shi** [reply]
Hm, if you `set` a value, you will `get` the value immediately, so you wouldn't need `forceAwait` at all. Please give it a try.

**Reply by misaku** [reply]
i try to check and thats works!
in state of react if i don make force await, some cases, my application dont work correctly.
i will try to use in my application  that have this problem and try to check if i will have same problem.
thaks to help me
